pkg/agent/controller/traceflow/traceflow_controller.go

```diff
@@ -118,6 +124,11 @@ func NewTraceflowController(
 	)
 	// Register packetInHandler
 	c.ofClient.RegisterPacketInHandler("traceflow", c)
+	// Add serviceLister if AntreaProxy enabled
+	if features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
+		c.serviceLister = informerFactory.Core().V1().Services().Lister()
```

Comment:
Do we need to use Informer? Could we just Get the Service on demand?

Reply:
Yes. We need to use Lister to get the Service from Cache.
We can just get the Service using Client, but to get Service from Cache is a better choice.
This Informer will be enabled only when AntreaProxy is enabled, thus no additional performance cost.

Reply:
Ok.